xds/src/main/java/io/grpc/xds/sds/TlsCertificateSecretVolumeSecretProvider.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *  Implementation of a file based secret provider.
+ */
+final class TlsCertificateSecretVolumeSecretProvider
+    implements SecretProvider<TlsCertificateStore> {
+
+  private static final Logger logger = Logger
+      .getLogger(TlsCertificateSecretVolumeSecretProvider.class.getName());
+  public static final String PEM = ".pem";
+  public static final String CRT = ".crt";
+
+  private String path;
+
+  // for now mark it unused
+  @SuppressWarnings("unused")
+  private String name;
```

Comment:
These fields are final or can be changed?

Reply:
good point. Made final